hack/update-gofmt.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 The Kubernetes Authors.
```

Comment:
I think we should update the year to 2023, not sure though.
thoughts @jayunit100 @knabben ?

Rest all looks good and matches k/k upstream!

Reply:
Thanks for the review Amim and Daman updated the date